### PR TITLE
Pass date and time format to timeline forms

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26178

Before https://github.com/metabase/metabase/pull/26247 and https://github.com/metabase/metabase/pull/26289 `dateFormat` and `timeFormat` were passed by the "form framework" to the underlying date field. Now it needs to be passed manually. This PR adds types for the formatting settings and passes them to `EventForm`.  It also adds e2e tests to make sure we won't break this feature in the future.

### CODE REVIEW

File: frontend/src/metabase-types/api/settings.ts [hunk 1]
```diff
@@ -1,8 +1,32 @@
+export interface FormattingSettings {
+  "type/Temporal"?: DateFormattingSettings;
```
Comment (ranquild) [inline frontend/src/metabase-types/api/settings.ts:2]: This is just our existing API contract for the settings, it has nothing to do with semantic types.

File: frontend/test/metabase/scenarios/organization/timelines-collection.cy.spec.js [hunk 2]
```diff
@@ -97,8 +98,7 @@ describe("scenarios > organization > timelines > collection", () => {
 
       cy.findByLabelText("Event name").type("RC1");
 
-      // New event modal
-      cy.get(".Modal").within(() => {
+      getModal().within(() => {
```
Comment (kulyk) [inline frontend/test/metabase/scenarios/organization/timelines-collection.cy.spec.js:101]: fyi, there is also a [shared `modal` selector](https://github.com/metabase/metabase/blob/53effa4a5e44529e49b6cba936054248d2c5e3fe/frontend/test/__support__/e2e/helpers/e2e-ui-elements-helpers.js#L10) for e2e tests

File: frontend/src/metabase/core/components/TimeInput/TimeInput.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ import {
 
 export interface TimeInputProps {
   value: Moment;
-  is24HourMode?: boolean;
+  timeFormat?: string;
```
Comment (kulyk) [inline frontend/src/metabase/core/components/TimeInput/TimeInput.tsx:18]: I wonder what format will be used when it's not explicitly set?
Should we maybe set a default now, so it's a bit more clear?

File: frontend/src/metabase/timelines/common/containers/EventForm/EventForm.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { connect } from "react-redux";
+import { State } from "metabase-types/store";
+import EventForm, {
+  EventFormOwnProps,
+  EventFormStateProps,
+} from "../../components/EventForm";
+
+const mapStateToProps = (state: State) => ({
+  formattingSettings: state.settings.values["custom-formatting"],
```
Comment (kulyk) [inline frontend/src/metabase/timelines/common/containers/EventForm/EventForm.tsx:9]: Think we should use the [`getSetting` selector](https://github.com/metabase/metabase/blob/53effa4a5e44529e49b6cba936054248d2c5e3fe/frontend/src/metabase/selectors/settings.js#L5) here?

File: frontend/src/metabase/timelines/common/components/EventForm/EventForm.tsx [hunk 5]
```diff
@@ -29,22 +33,31 @@ const EventSchema = Yup.object({
...
   onArchive?: () => void;
   onCancel?: () => void;
 }
 
+export interface EventFormStateProps {
+  formattingSettings?: FormattingSettings;
+}
+
+export type EventFormProps = EventFormOwnProps & EventFormStateProps;
+
```
Comment (kulyk) [inline frontend/src/metabase/timelines/common/components/EventForm/EventForm.tsx:50]: Seems like `formattingSettings` are defined in both `OwnProps` and `StateProps` now. Can we keep only one?

**Review by kulyk — APPROVED** [review]
Looks good! Please take a look at the comments, especially ones about settings selector and props duplication

File: frontend/src/metabase/timelines/common/components/EventForm/EventForm.tsx (hunk 5 above)
Comment (ranquild) [reply frontend/src/metabase/timelines/common/components/EventForm/EventForm.tsx]: Fixed

File: frontend/src/metabase/timelines/common/containers/EventForm/EventForm.tsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/timelines/common/containers/EventForm/EventForm.tsx]: Fixed, didn't know about it

File: frontend/src/metabase/core/components/TimeInput/TimeInput.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/core/components/TimeInput/TimeInput.tsx:19]: Fixed

File: frontend/test/metabase/scenarios/organization/timelines-collection.cy.spec.js (hunk 2 above)
Comment (ranquild) [reply frontend/test/metabase/scenarios/organization/timelines-collection.cy.spec.js:101]: It's a custom modal so this helper doesn't work

**Pull request merged**